src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -682,6 +688,89 @@ class spell_warl_life_tap : public SpellScriptLoader
         }
 };
 
+// -30881 - Nature's Guardian
+class spell_warl_nether_protection : public SpellScriptLoader
+{
+public:
+    spell_warl_nether_protection() : SpellScriptLoader("spell_warl_nether_protection") { }
+
+    class spell_warl_nether_protection_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warl_nether_protection_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_HOLY)
+                || !sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_FIRE)
+                || !sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_FROST)
+                || !sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_ARCANE)
+                || !sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_SHADOW)
+                || !sSpellMgr->GetSpellInfo(SPELL_WARLOCK_NETHER_PROTECTION_NATURE))
+                return false;
+            return true;
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            if (eventInfo.GetDamageInfo())
+                switch (GetFirstSchoolInMask(eventInfo.GetDamageInfo()->GetSchoolMask()))
+                {
```

Comment:
its not needed - it was returning false... should only return true when a magic spell school is used